lib/utils/process-jobs.js

```diff
@@ -160,7 +160,9 @@ module.exports = function(extraJob) {
     // Lock the job in MongoDB!
     self._collection.findOneAndUpdate(criteria, update, options, (err, resp) => {
       if (err) {
-        throw err;
+        debug('job lock failed while locking on the fly');
+        self.emit('error', err);
+        return lockOnTheFly();
```

Comment:
Why the retry?

Reply:
I removed this retry to keep this PR more aligned with the original behaviour.

If it's needed, we can loop back in a separate PR.